weave/trace_server/calls_query_builder.py

```diff
@@ -1378,29 +1445,132 @@ def process_operand(operand: tsi_query.Operand) -> Optional[str]:
                 tsi_query.ContainsOperation,
             ),
         ):
-            processed = process_operation(operand)
-            if processed is None:
-                return None
-            return processed.str_filter_opt_sql
+            return process_operation(operand)
         return None
 
     # Create a single AND operation from all conditions
-    and_operation = tsi_query.AndOperation(
-        **{"$and": [c.operand for c in filterable_conditions]}
-    )
+    and_operation = tsi_query.AndOperation(**{"$and": [c.operand for c in conditions]})
 
     # Process the combined operation
     processed = process_operation(and_operation)
     if processed is None:
         return OptimizationConditions()
-    if not processed.str_filter_opt_sql:
+
+    if not processed.str_filter_opt_sql and not processed.id_datetime_filters_sql:
         return OptimizationConditions()
 
-    processed.str_filter_opt_sql = "AND " + processed.str_filter_opt_sql
+    if processed.str_filter_opt_sql:
+        processed.str_filter_opt_sql = "AND " + processed.str_filter_opt_sql
+    if processed.id_datetime_filters_sql:
+        # Create non-uuidv7 condition
+        non_uuidv7_condition = f"{table_alias}.id <= 'ffffffffffffffff'"
```

Comment:
I think this `non-uuid7` condition should be appended inside of the `_create_datetime_optimization_sql` procedure so this caller does not need to know about those details

Reply:
I agree its ugly, I can play around with it. The reason its there is that if you have multiple id conditions, like "x > 111 and y < 1111" we only ever want one `id < ffffffff`. Thats fine for this example, but what if the datetime condition is deep in a nested and + or + in stack we still just only want the one condition at the top (if we ever touch an id pre-where, always always let these kinds of ids pass). I will figure out how to make it prettier, the recursion is annoying hmm.